car-bridge/src/streaming.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+use std::{
+    collections::{hash_map::Entry, HashMap, HashSet},
+    sync::Arc,
+};
+
+use chariott_common::{chariott_api::ChariottCommunication, error::Error};
+use chariott_proto::streaming::Event;
+use ess::{EventSubSystem, NotReadingEvents};
+use examples_common::chariott::api::{Chariott, ChariottCommunicationExt as _};
+use prost::Message as _;
+use tokio::spawn;
+use tokio_stream::StreamExt as _;
+use tracing::{debug, warn};
+
+use crate::{Message, ResponseSender};
+
+/// Identifies a namespace
+type Namespace = String;
+/// Identifies a topic
+type Topic = String;
+/// Identifies a subscription source (a.k.a. _key_).
+type Source = String;
```

Comment:
nit: Want to call it `SubscriptionSource` so the comment isn't needed?

Reply:
done in 7a0eb903e02d9c3457de4649e8f5cbdbe3b087e7